testplan/common/entity/base.py

```diff
@@ -86,6 +92,14 @@ def remove(self, uid):
     def first(self):
```

Comment:
I think this method is used to get the most frequently used resource, but not always local runner (something else can also executes test). I agree that it's not a good assumption but removing it will cause a lot of other changes so I will not do it in this PR.

Reply:
removed properties and in our code use "parent.abc" to access it.